### ISSUE TITLE
MacOS Installation Won't Work Faderport 8

### ISSUE THREAD

**Issue by BoilerSnake** [navelpluisje/reasonus-install#42]
**Describe the bug**
ReaSonus installer launches fine, but doesn't present me with the option to select the midi device, only presents me with `00`

I believe that this is the same bug that fuux84 encountered, so I will attach my reaper-midihw.ini file. Until then, ReaSonus is sadly not working.

**Operating System**
- [] Mac OS

**Versions**
Versions can be found on the About page of the installer
- Reasonus Installer: Darwin Universal v1.0.1
- Reasonus Resources: v1.0.1
- CSI (ReaSonus): 1.1 (reasonus-edition)

**Expected behavior**
Should give me the option to select my faderport device to initiate installation, but it does not.

**Additional context**
Reaper-midihw.ini: _I have changed the file extension to TXT so I can upload it to github_
[reaper-midihw copy.txt](https://github.com/navelpluisje/reasonus-install/files/11407167/reaper-midihw.copy.txt)

**Comment by navelpluisje** [comment]
Hmmm, was checking your file, and it looks like something has changed in the naming. I'm not at home currently. Will ba able to check it from Wednesday on.
Can you try to unplug the faderport, delete thos file, restart Reaper and reconnect the faderport again? That's for now the only thing I can think of.

**Comment by BoilerSnake** [comment]
No such luck I'm afraid, didn't see any difference with the installer. I've got a brand new Faderport here myself from Thomann just last week.

Uploading the new midihw after the restart in case it helps! And hey, thanks for the help chief! Incredible work you're doing <3
[reaper-midihw restarted.txt](https://github.com/navelpluisje/reasonus-install/files/11409054/reaper-midihw.restarted.txt)

**Comment by BoilerSnake** [comment]
I managed to install ReaSonus on my Win10-64 PC, but running CSI as the DAW control interface gave me some weird bugs. Selecting track mode did nothing, just set the faders for -60dB regardless of what the track was set to. Selecting edit plugins would set faders to match the tracks, then pressing solo would select a track where select does nothing... The screen would reveal nothing.
I don't know what's going on, it's odd. As it stands Mackie Control Universal worked more accurately. Anything I could do to help troubleshoot this?

**Comment by navelpluisje** [comment]
In what mode is your faderport? It needs to be in Studio 1 mode. How to set this can be found in the manual at page 6

**Comment by BoilerSnake** [comment]
Yeah, I booted it up on the PC today and it was running fine so that was my own user-error (being a dumbass has it's downsides). Works great now!

As for the Mac, still have no luck in getting it installed sadly

**Comment by navelpluisje** [comment]
Nice it's working on Windows.
For the mac, can you remove the midihw file, reboot REAPER and reconnect the controller? 
It should not break anything because there is only one device in the list.

In the midi devices should be something with these names:
<img width="697" alt="image" src="https://github.com/navelpluisje/reasonus-install/assets/1625221/4418c644-ac0e-43ba-9344-4edbe5429c47">

**Comment by BoilerSnake** [comment]
I've done that, it reads correctly as a midi device, but it still does not let me install ReaSonus on Mac I'm afraid, giving me the same 00 message. Running MacOS Monterey 12.5 if that helps at all.

[reaper-midihwDeleted.txt](https://github.com/navelpluisje/reasonus-install/files/11445839/reaper-midihwDeleted.txt)

**Comment by navelpluisje** [comment]
Mist this massage. Do you also have the port 1 in the list? This one is not showing up in the txt file. That's the port that should normally be used for the faderport

**Comment by BoilerSnake** [comment]
I set Faderport to ID: 1 in MIDI devices if that is what you mean? Didn't change no matter what ID I set

**Comment by BoilerSnake** [comment]
Oh heck, I see what you meant now. No, in Midi Devices neither Port 1 NOR Midi Mode turn up. On Windows, MidiIn/MidiOut showed up, but it doesn't on Mac even when I put it in Midi mode :/

**Comment by BoilerSnake** [comment]
I've figured out the issue! Turns out it was entirely on my end but unassumingly so. 

The MIDIin/MIDIout ports are available on newer versions of the firmware so far as I can tell: https://forums.presonus.com/viewtopic.php?t=38081

As I turned on my Faderport on an older version of the firmware, I suspect that the firmware didn't have this MidiIn/Out feature. Deleting the FP8 device in Audio-Midi Setup/MIDI studio fixed it. When the device was reconnected, the MIDI ports showed up. I would personally recommend adding this set up to the guide should you feel so inclined.

Installed fine, but after installation Reaper would not boot and had to manually uninstall the plugin to get it to launch again, so I'm gonna part using Reasonus on MacOS _for now._ Keep using it on Windows, incredible piece of software!
